modules/sync/syncs/httpsynchronization.cpp

```diff
@@ -134,6 +158,82 @@ std::string HttpSynchronization::generateUid() {
     return fmt::format("{}/{}", _identifier, _version);
 }
 
+void HttpSynchronization::createSyncFile(bool fullySynchronized = true) const {
+    std::filesystem::path dir = directory();
+    std::filesystem::create_directories(dir);
+
+    dir.replace_extension("ossync");
+    std::ofstream syncFile(dir, std::ofstream::out);
+
+    syncFile << _ossyncVersionNumber << '\n' <<
+        (fullySynchronized ? _synchronizationToken : "Partial Synchronized") << '\n';
+
+    if (fullySynchronized)
+        return; //All files successfully downloaded, no need to write anything else to file.
+
+    //Store all files that successfully downloaded
+    for (std::string const& fileURL : _existingSyncedFiles) {
+        syncFile << fileURL << '\n';
+    }
+    //If we we fill _existingSyncedFiles before calling this func this loop is uneccessary
+    for (std::string const& fileURL : _newSyncedFiles) {
+        syncFile << fileURL << '\n';
+    }
+}
+
+bool HttpSynchronization::checkSyncFile() {
+    std::filesystem::path path = directory();
+    path.replace_extension("ossync");
+    //Check if file exists at all
+    if (!std::filesystem::is_regular_file(path)) {
+        return false;
+    }
+
+    //Read contents of file
+    std::ifstream file{ path };
+    std::string line;
+    std::string ossyncVersion;
+
+    file >> line;
+    //Ossync files that does not have a version number are already resolved
+    //As they are of the previous format.
+    if (line == "Synchronized" /*_synchronizationToken*/) {
+        return true;
+    }
+    //Otherwise first line is the version number.
+    ossyncVersion = line;
+
+    /*
+    Format of 1.0 ossync:
+    Version number: E.g., 1.0
+    Synchronization status: Synchronized or Partial Synchronized
+    Optionally list of already synched files
+    */
+
+    if (ossyncVersion == "1.0" /*_ossyncVersionNumber*/) {
+        std::getline(file >> std::ws, line); //Read synchronization status
+        if (line == _synchronizationToken) {
+            return true;
+        }
+        //File is only partially synchronized,
+        // store file urls that have been synched already. 
+        while (file >> line) {
+            if (line.empty() || line[0] == '#') {
+                // Skip all empty lines and commented out lines
+                continue;
+            }
+            _existingSyncedFiles.push_back(line);
+        }
+    }
+    else {
+        LWARNING(fmt::format("{}: Unknown ossync version number read."
+            "Got{} while {} and below are valid!"
+            , _identifier, ossyncVersion, _ossyncVersionNumber));
```

Comment:
Message macros together with fmt::format call are one of the few exceptions to the "one function be line" rule, so this would be:

```suggestion
        LWARNING(fmt::format(
            "{}: Unknown ossync version number read."
            "Got {} while {} and below are valid!",
            _identifier,
            ossyncVersion,
            _ossyncVersionNumber
        ));
```